### PR TITLE
Annotate more files with nullable enable

### PR DESCRIPTION
**Pull request by Therzok**

### CODE REVIEW

**Review by mrward — APPROVED** [review]

**Review by iainx — APPROVED** [review]

File: main/src/core/MonoDevelop.Core/MonoDevelop.Core/ProgressMonitor.cs [hunk 1]
```diff
@@ -678,7 +693,7 @@ protected virtual void OnWarningReported (string message)
 		{
 		}
 
-		protected virtual void OnErrorReported (string message, Exception exception)
+		protected virtual void OnErrorReported (string? message, Exception? exception)
```
Comment (slluis) [inline main/src/core/MonoDevelop.Core/MonoDevelop.Core/ProgressMonitor.cs]: Is this an api break?

Comment (Therzok) [reply main/src/core/MonoDevelop.Core/MonoDevelop.Core/ProgressMonitor.cs]: No

Comment (Therzok) [reply main/src/core/MonoDevelop.Core/MonoDevelop.Core/ProgressMonitor.cs]: It's basically just an annotation that gets transformed into an attribute in the dll.

Comment (Therzok) [reply main/src/core/MonoDevelop.Core/MonoDevelop.Core/ProgressMonitor.cs]: https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+BiAdgVwDZ4CGweMABDFsaQLABQ9AAgEwCMTAzGS2QMJkBvemRHcujFGQCyrABSNWABjIIAlMNFC6osgF8NIg2O6SpzeUoD8K9ds1H9dXUA

**Review by slluis — APPROVED** [review]

**Pull request merged**